lab_5_scraper/scraper.py

```diff
@@ -19,200 +73,241 @@ def __init__(self, path_to_config: pathlib.Path) -> None:
 
-        Returns:
-            int: Number of seconds to wait for response
-        """
+    def get_num_articles(self) -> int:
+        return self._num_articles
 
-    def get_verify_certificate(self) -> bool:
-        """
-        Retrieve whether to verify certificate.
+    def get_headers(self) -> dict[str, str]:
+        return self._headers
 
-        Returns:
-            bool: Whether to verify certificate or not
-        """
+    def get_encoding(self) -> str:
+        return self._encoding
 
-    def get_headless_mode(self) -> bool:
-        """
-        Retrieve whether to use headless mode.
+    def get_timeout(self) -> int:
+        return self._timeout
 
-        Returns:
-            bool: Whether to use headless mode or not
-        """
+    def get_verify_certificate(self) -> bool:
+        return self._should_verify_certificate
+
+    def get_headless_mode(self) -> bool:
+        return self._headless_mode
 
 
 def make_request(url: str, config: Config) -> requests.models.Response:
     """
-    Deliver a response from a request with given configuration.
-
-    Args:
-        url (str): Site url
-        config (Config): Configuration
-
-    Returns:
-        requests.models.Response: A response from a request
+    Make HTTP request to specified URL with given configuration.
     """
+    try:
+        response = requests.get(
+            url,
+            headers=config.get_headers(),
+            timeout=config.get_timeout(),
+            verify=config.get_verify_certificate()
+        )
+        response.encoding = config.get_encoding()
+        return response
+    except requests.exceptions.RequestException as e:
+        print(f"Error making request to {url}: {e}")
+        raise
 
 
 class Crawler:
     """
-    Crawler implementation.
+    Crawler implementation for Mordovia News website.
     """
 
-    #: Url pattern
-    url_pattern: Union[Pattern, str]
+    url_pattern = re.compile(r'news-\d-\d{5}.htm')
 
     def __init__(self, config: Config) -> None:
-        """
-        Initialize an instance of the Crawler class.
-
-        Args:
-            config (Config): Configuration
-        """
+        self.config = config
+        self.urls = []
+        prepare_environment(ASSETS_PATH)
 
     def _extract_url(self, article_bs: BeautifulSoup) -> str:
         """
-        Find and retrieve url from HTML.
-
-        Args:
-            article_bs (bs4.BeautifulSoup): BeautifulSoup instance
-
-        Returns:
-            str: Url from HTML
+        Extract article URL from BeautifulSoup object.
         """
+        article_link = article_bs.find_all('dd', {'class': 'title'})
+        for url in article_link:
+            url = url.find('a')
+            if 'https://mordovia-news.ru/' + url['href'] not in self.urls:
+                return 'https://mordovia-news.ru/' + url['href'] if article_link else ''
+        return ''
 
     def find_articles(self) -> None:
         """
-        Find articles.
```

Comment:
Please, do not modify docstrings, this is a part of the code along with the methods declarations and class definitions.